packages/shared/src/utils.ts

```diff
@@ -182,3 +183,14 @@ export function getDims(dataArray: NdArray): Dims {
   const [rows, cols] = dataArray.shape;
   return { rows, cols };
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function createMemo<P extends any[], R, T extends (...args: P) => R>(
+  fn: T
+): T;
```

Comment:
I need an override to explicitly declare `T` as return type. If I declare this return type on the function itself, I get an error: `'(...args: P) => R' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint '(...args: P) => R'.`